app/src/ui/diff/text-diff-expansion.ts

```diff
@@ -49,12 +49,17 @@ function mergeDiffHunks(hunk1: DiffHunk, hunk2: DiffHunk): DiffHunk {
     false
   )
 
+  const newHunkLines = [
+    newFirstHunkLine,
+    ...allHunk1LinesButFirst,
+    ...allHunk2LinesButFirst,
+  ]
+
   return new DiffHunk(
     newHunkHeader,
-    [newFirstHunkLine, ...allHunk1LinesButFirst, ...allHunk2LinesButFirst],
+    newHunkLines,
     hunk1.unifiedDiffStart,
-    // This -1 represents the header line of the second hunk that we removed
-    hunk2.unifiedDiffEnd - 1,
```

Comment:
https://github.com/desktop/desktop/blob/8daf7ede719fd0d013084ff8ef5bac1930bfe8c9/app/src/ui/diff/text-diff-expansion.ts#L311-L335

This `hunk2` is the `adjacentHunk` you see in line 334. However, whereas `hunk1` had the right `unifiedDiffStart` and `unifiedDiffEnd` after expansion (see lines 315-316), the `adjacentHunk` is never updated like that.